foundationdb/tests/test_foundationdb.py

```diff
@@ -1,15 +1,87 @@
+    "foundationdb.workload.transactions.started_batch_priority.hz",
+    "foundationdb.workload.transactions.started_default_priority.counter",
+    "foundationdb.workload.transactions.started_default_priority.hz",
+    "foundationdb.workload.transactions.started_immediate_priority.counter",
+    "foundationdb.workload.transactions.started_immediate_priority.hz",
+    "foundationdb.workload.operations.location_requests.counter",
+    "foundationdb.workload.operations.location_requests.hz",
+    "foundationdb.workload.operations.low_priority_reads.counter",
+    "foundationdb.workload.operations.low_priority_reads.hz",
+    "foundationdb.workload.operations.memory_errors.counter",
+    "foundationdb.workload.operations.memory_errors.hz",
+    "foundationdb.workload.operations.read_requests.counter",
+    "foundationdb.workload.operations.read_requests.hz",
+    "foundationdb.workload.operations.reads.counter",
+    "foundationdb.workload.operations.reads.hz",
+    "foundationdb.workload.operations.writes.counter",
+    "foundationdb.workload.operations.writes.hz",
+    "foundationdb.data.system_kv_size_bytes",
+    "foundationdb.data.total_disk_used_bytes",
+    "foundationdb.data.total_kv_size_bytes",
+    "foundationdb.data.least_operating_space_bytes_log_server",
+    "foundationdb.data.moving_data.in_flight_bytes",
+    "foundationdb.data.moving_data.in_queue_bytes",
+    "foundationdb.data.moving_data.total_written_bytes",
+    "foundationdb.datacenter_lag.seconds",
+    "foundationdb.instances",
+]
 
-def test_check(aggregator, instance):
+current_dir = dir_path = os.path.dirname(os.path.realpath(__file__)) + '/'
+def test_partial(aggregator, instance):
+    with open(current_dir + 'partial.json', 'r') as f:
+        data = json.loads(f.read())
+        check = FoundationdbCheck('foundationdb', {}, [instance])
+        check.check(instance)
+        aggregator.assert_service_check("foundationdb.can_connect", AgentCheck.OK)
+
+def test_full(aggregator, instance):
+    with open(current_dir + 'full.json', 'r') as f:
+        data = json.loads(f.read())
+        check = FoundationdbCheck('foundationdb', {}, [instance])
+        check.check_metrics(data)
+
+        for metric in METRICS:
+            aggregator.assert_metric(metric)
+        aggregator.assert_all_metrics_covered()
+        aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+        aggregator.assert_service_check("foundationdb.can_connect", AgentCheck.OK)
+
+
+def test_integ(aggregator, instance):
     # type: (AggregatorStub, Dict[str, Any]) -> None
     check = FoundationdbCheck('foundationdb', {}, [instance])
     check.check(instance)
 
+    for metric in METRICS:
+        aggregator.assert_metric(metric)
     aggregator.assert_all_metrics_covered()
     aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+    aggregator.assert_service_check("foundationdb.can_connect", AgentCheck.OK)
```

Comment:
This relies on `fdbcli` being installed, and so is the test we could probably upgrade to use the Docker containers?

Reply:
(Not that doing said work is a task for this PR, to be clear. Just making sure I understand correctly.)

Reply:
Yes, that's the idea